src/coreclr/jit/forwardsub.cpp

```diff
@@ -0,0 +1,622 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "jitpch.h"
+
+#ifdef _MSC_VER
+#pragma hdrstop
+#endif
+
+// Simple Forward Substitution
```

Comment:
Here would be a good place for a nice long comment:
1. Describing what "forward substitution" means
2. Explains why we need it (e.g., describe if we're "fixing" importer decisions or other phase choices)
3. Explains why it's in this place in the phase order
4. Gives examples
5. Says what cases are handled and not handled (and says whether they are fundamental or implementation limitations)
6. Gives a rough outline of the algorithm

Perhaps some of these could/should be elsewhere, closer to the code that does the work, but putting a reference / link here to that would also be useful.

Reply:
Assuming what I wrote suffices....